ci-operator/step-registry/baremetal/lab/agent/day2/baremetal-lab-agent-day2-commands.sh

```diff
@@ -0,0 +1,261 @@
+      "Waiting ${period}min before retrying (timeout in $(( (max_retries - i) * (period) ))min)..."
+  done
+}
+
+function monitor_workers()
+{
+  local DAY2_IPS=""
+
+  # shellcheck disable=SC2154
+  for bmhost in $(yq e -o=j -I=0 '.[]' "${SHARED_DIR}/hosts.yaml"); do
+    # shellcheck disable=SC1090
+    . <(echo "$bmhost" | yq e 'to_entries | .[] | (.key + "=\"" + .value + "\"")')
+    if [[ "${name}" == *-a-* ]] && [ "${ADDITIONAL_WORKERS_DAY2}" == "true" ]; then
+      DAY2_IPS+="${ip},"
+    fi
+  done
+  DAY2_IPS=${DAY2_IPS%,}
+  /tmp/oc adm node-image monitor --ip-addresses "${DAY2_IPS}" -a "${day2_pull_secret}"
+}
+
+SSHOPTS=(-o 'ConnectTimeout=5'
+  -o 'StrictHostKeyChecking=no'
+  -o 'UserKnownHostsFile=/dev/null'
+  -o 'ServerAliveInterval=90'
+  -o LogLevel=ERROR
+  -i "${CLUSTER_PROFILE_DIR}/ssh-key")
+
+#BASE_DOMAIN=$(<"${CLUSTER_PROFILE_DIR}/base_domain")
+#PULL_SECRET_PATH=${CLUSTER_PROFILE_DIR}/pull-secret
+
+
+INSTALL_DIR="${INSTALL_DIR:-/tmp/installer}"
+mkdir -p "${INSTALL_DIR}"
+day2_pull_secret="${INSTALL_DIR}/day2_pull_secret"
+
+
+cat > "${INSTALL_DIR}/nodes-config.yaml" <<EOF
+hosts: []
+EOF
+
+# shellcheck disable=SC2154
+for bmhost in $(yq e -o=j -I=0 '.[]' "${SHARED_DIR}/hosts.yaml"); do
+  # shellcheck disable=SC1090
+  . <(echo "$bmhost" | yq e 'to_entries | .[] | (.key + "=\"" + .value + "\"")')
+  if [[ "${name}" == *-a-* ]] && [ "${ADDITIONAL_WORKERS_DAY2}" == "true" ]; then
+
+  ADAPTED_YAML="
+  hostname: ${name}
+  interfaces:
+  - macAddress: ${mac}
+    name: ${baremetal_iface}
+  networkConfig:
+    interfaces:
+    - name: ${baremetal_iface}
+      type: ethernet
+      state: up
+      ipv4:
+        enabled: ${ipv4_enabled}
+        dhcp: false
+        address:
+            - ip: ${ip}
+              prefix-length: ${INTERNAL_NET_CIDR##*/}
+      ipv6:
+        enabled: ${ipv6_enabled}
+"
+
+  # Take care of the indentation when adding the dns and routes to the above yaml
+  ADAPTED_YAML+="
+    dns-resolver:
+          config:
+            server:
+              - ${INTERNAL_NET_IP}
+    routes:
+      config:
+        - destination: 0.0.0.0/0
+          next-hop-address: ${INTERNAL_NET_IP}
+          next-hop-interface: ${baremetal_iface}
+  "
+  # Patch the nodes-config.yaml by adding the given host to the hosts list in the platform.baremetal stanza
+  yq --inplace eval-all 'select(fileIndex == 0).hosts += select(fileIndex == 1) | select(fileIndex == 0)' \
+    "$INSTALL_DIR/nodes-config.yaml" - <<< "$ADAPTED_YAML"
+  fi
+done
+
+cp "${INSTALL_DIR}/nodes-config.yaml" "${ARTIFACT_DIR}/"
+
+export KUBECONFIG="$SHARED_DIR/kubeconfig"
+export http_proxy="${proxy}" https_proxy="${proxy}" HTTP_PROXY="${proxy}" HTTPS_PROXY="${proxy}"
+
+cat "${CLUSTER_PROFILE_DIR}/pull-secret" > "${day2_pull_secret}"
+
+echo "Extract the latest oc client..."
+oc adm release extract -a "${day2_pull_secret}" "${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE}" \
```

Comment:
You can add cli: latest in the ref.yaml file instead of this. Oc will be put in /cli/oc and /cli will be in the PATH

Reply:
Thanks, let me use cli: latest  instead...

Reply:
Hi @aleskandro , cli: latest version is not consistent with the nightly build running the job, I am wondering if it is ok since we are testing 'oc' in this step: 
```
sh-5.1$ /cli/oc version
Client Version: 4.17.0-202408121444.p0.g51ca91a.assembly.stream.el9-51ca91a
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3

sh-5.1$ /tmp/openshift-install version
/tmp/openshift-install 4.17.0-0.nightly-2024-08-15-153405
```